pipeline/pipeline/backend.py

```diff
@@ -283,16 +283,16 @@ def _cp(src, dst):
             return
 
         if delete_scratch_on_exit and used_remote_tmpdir:
-            parent_ids = list(job_id_to_command.keys())
+            parents = list(jobs_to_command.keys())
```

Comment:
I think this is a really cool interface decision; it seems we don't need custom __hash__ and __eq__... but want to confirm (Python docs state that default __hash__ is id(), which is a stable identifier that is unique for the lifetime of the object).

Reply:
I'm pretty sure this is fine. https://stackoverflow.com/questions/4950155/objects-as-keys-in-python-dictionaries?rq=1